src/utils.c

```diff
@@ -1748,3 +1750,65 @@ ts_update_placeholder(PG_FUNCTION_ARGS)
 	elog(ERROR, "this stub function is used only as placeholder during extension updates");
 	PG_RETURN_NULL();
 }
+
+/*
+ * Check if a relation is using hyperstore AM.
+ *
+ * Assumes that the relation already has appropriate locks.
+ */
+bool
+ts_relation_uses_hyperstore(Oid relid)
+{
+#if WITH_HYPERSTORE
+	HeapTuple tup;
+	Oid clamoid = InvalidOid;
+	Oid amoid = get_am_oid("hyperstore", false);
+
+	tup = SearchSysCache1(RELOID, ObjectIdGetDatum(relid));
```

Comment:
Noticed this catalog access in the profiles. Should this be cached in the Chunk struct and filled together with `reltype` in a single catalog lookup?

Reply:
This can also be merged separately with non-existent `hyperstore` AM and the `get_am_oid` returning `InvalidOid`, by the way.

Reply:
Surprised that this would make such a difference. It is using the SysCache, so it _is_ a cached value. Sure, there could have been a cache invalidation, but not sure why that would happen during planning...

Reply:
> Noticed this catalog access in the profiles. Should this be cached in the Chunk struct and filled together with `reltype` in a single catalog lookup?

Sure, we can do that and it doesn't hurt. But this function is often called when you don't have a chunk and perhaps you don't want the overhead of building a chunk. Chunk lookups are not cached.

Reply:
> But this function is often called when you don't have a chunk and perhaps you don't want the overhead of building a chunk.

Where? I saw about 5 uses and each has a Chunk struct nearby. Maybe we could have a separate function for non-chunk cases, but when there's a Chunk already, better use it.


> Chunk lookups are not cached.

Right, we shouldn't do chunk lookups, we made a lot of effort to systematically cache the Chunk struct in the `fdw_private` to avoid doing them.